tests/Api/Shop/PayingForOrdersByPblTest.php

```diff
@@ -0,0 +1,135 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Api\Shop;
+
+use Sylius\Component\Core\Model\OrderInterface;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Tests\CommerceWeavers\SyliusTpayPlugin\Api\JsonApiTestCase;
+use Tests\CommerceWeavers\SyliusTpayPlugin\Api\Utils\CardEncrypterTrait;
+use Tests\CommerceWeavers\SyliusTpayPlugin\Api\Utils\OrderPlacerTrait;
+
+class PayingForOrdersByPblTest extends JsonApiTestCase
+{
+    use CardEncrypterTrait;
+    use OrderPlacerTrait;
+
+    protected function setUp(): void
+    {
+        parent::setUp();
+
+        $this->setUpOrderPlacer();
+    }
+
+    public function test_it_returns_violation_if_bank_is_not_available(): void
+    {
+        $this->loadFixturesFromDirectory('shop/paying_for_orders_by_card');
+
+        $order = $this->doPlaceOrder('t0k3n', paymentMethodCode: 'tpay_pbl');
+
+        $this->client->request(
+            Request::METHOD_POST,
+            sprintf('/api/v2/shop/orders/%s/pay', $order->getTokenValue()),
+            server: self::CONTENT_TYPE_HEADER,
+            content: json_encode([
+                'successUrl' => 'https://example.com/success',
+                'failureUrl' => 'https://example.com/failure',
+                'tpayChannelId' => '3'
+            ]),
+        );
+
+        $response = $this->client->getResponse();
+
+        $this->assertResponseViolations($response, [
+            [
+                'propertyPath' => 'tpayChannelId',
+                'message' => 'Channel with provided id is not available.',
+            ]
+        ]);
+    }
+
+    public function test_it_returns_violation_if_channel_is_not_a_bank(): void
```

Comment:
I'm not sure, but we talked we're not going to filter banks, at least for now.

Reply:
In case of Banks it needs to be a proper tpay channel (Bank), otherwise user would be making a payment to other type (f.e. user could make a payment to Visa Mobile, "druk" payment type :D or something else). So IMO the violation (validation) is necessary in this case.

Reply:
You might mismatch a returned list of Tpay channels where you can make a payment (the one from available channels, or TpayBank resource on our API as it will be named), and this Paying Action where you need to have a correct tpay channel id.